### DISCUSSION TITLE
Issue with reading OGC WKT given in CF-convention specifications

### DISCUSSION THREAD

**Discussion by observingClouds** [pyproj4/pyproj#1457] [Q&A] [answered]
Hi everyone 👋 ,

I'm not an expert in CRS definitions and am wondering if the following WKT is a valid OGC-WKT?

```
COMPOUNDCRS ["OSGB 1936 / British National Grid + ODN", PROJCRS ["OSGB 1936 / British National Grid", GEODCRS ["OSGB 1936", DATUM ["OSGB 1936", ELLIPSOID ["Airy 1830", 6377563.396, 299.3249646, LENGTHUNIT["metre",1.0]], TOWGS84[375, -111, 431, 0, 0, 0, 0]], PRIMEM ["Greenwich", 0], UNIT ["degree", 0.0174532925199433]], CONVERSION["OSGB", METHOD["Transverse Mercator", PARAMETER["False easting", 400000, LENGTHUNIT["metre",1.0]], PARAMETER["False northing", -100000, LENGTHUNIT["metre",1.0]], PARAMETER["Longitude of natural origin", -2.0, ANGLEUNIT["degree",0.0174532925199433]], PARAMETER["Latitude of natural origin", 49.0, ANGLEUNIT["degree",0.0174532925199433]], PARAMETER["Longitude of false origin", -7.556, ANGLEUNIT["degree",0.0174532925199433]], PARAMETER["Latitude of false origin", 49.766, ANGLEUNIT["degree",0.0174532925199433]], PARAMETER["Scale factor at natural origin", 0.9996012717, SCALEUNIT["Unity",1.0]], AUTHORITY["EPSG", "27700"]] CS[Cartesian,2], AXIS["easting (X)",east], AXIS["northing (Y)",north], LENGTHUNIT["metre", 1.0]], VERTCRS ["Newlyn", VDATUM ["Ordnance Datum Newlyn", 2005], AUTHORITY ["EPSG", "5701"] CS[vertical,1], AXIS["gravity-related height (H)",up], LENGTHUNIT["metre",1.0]]]]
```

I got this string from the example 5.12 in the [CF-conventions](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#use-of-the-crs-well-known-text-format), so I assumed it would be correct.

However, when reading this string with `pyroj` I get a `pyproj.exceptions.CRSError`: (Internal Proj Error: proj_create: Missing BASEGEODCRS / BASEGEOGCRS / GEOGCS node)

Upon a quick inspection of the OGC WKT specifications [18-010r11](https://docs.ogc.org/is/18-010r11/18-010r11.pdf) I could not find any hint that those values are always required.

Your help would be highly appreciated.

Cheers!

**Comment by snowman2** [comment, marked as answer]
PROJ handles the WKT parsing logic. I recommend either asking on their mailing list or opening an issue if you believe there is a bug: https://github.com/osgeo/proj
